### PR TITLE
net:add btsnoop and snoop

### PR DESCRIPTION
**Pull request by chengkai15**
Signed-off-by: chengkai <chengkai@xiaomi.com>

## Summary
add btsnoop and snoop  feature

## Impact
NO

## Testing
1. wireshark   could load btsnoop pass
2. ellisys import btsnoop pass

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: net/utils/net_snoop.c [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+ ****************************************************************************/
+
+int snoop_sync(FAR struct snoop_s *snoop)
+{
+  if (!snoop)
+    {
+      return -EINVAL;
+    }
+
+  return file_fsync(&snoop->filep);
```
Comment (xiaoxiang781216) [inline net/utils/net_snoop.c:374]: add CONFIG_DISABLE_MOUNTPOINT check

File: include/nuttx/net/snoop.h [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+#define SNOOP_DATALINK_TYPE_FDDI        8
+#define SNOOP_DATALINK_TYPE_OTHER       9
+
+#define SNOOP_DATALINK_HCI_UNENCAP      1001
+#define SNOOP_DATALINK_HCI_UART         1002
+#define SNOOP_DATALINK_HCI_BSCP         1003
+#define SNOOP_DATALINK_HCI_SERIAL       1004
+
+#define SNOOP_DIRECTION_FLAG_SENT 0 /* Direction flag 0 = Sent */
+#define SNOOP_DIRECTION_FLAG_RECV 1 /* Direction flag 1 = Received */
```
Comment (acassis) [inline include/nuttx/net/snoop.h:82]: Please align it with above blocks

Comment (chengkai15) [reply include/nuttx/net/snoop.h]: > Please align it with above blocks

done

File: net/utils/net_snoop.c (hunk 1 above)
Comment (chengkai15) [reply net/utils/net_snoop.c]: > add CONFIG_DISABLE_MOUNTPOINT check

done

File: net/utils/net_snoop.c [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+
+  if (!snoop)
+    {
+      return -EINVAL;
+    }
+
+#ifndef CONFIG_DISABLE_MOUNTPOINT
+  ret = file_fsync(&snoop->filep);
+#else
+  ret = OK;
```
Comment (xiaoxiang781216) [inline net/utils/net_snoop.c:379]: merge to line 369.

Comment (chengkai15) [reply net/utils/net_snoop.c]: done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**